api/src/pcapi/routes/adage_iframe/offers.py

```diff
@@ -47,6 +48,32 @@ def get_educational_offers_categories(
     )
 
 
+@blueprint.adage_iframe.route("/collective/all-offers", methods=["GET"])
+@spectree_serialize(response_model=serializers.ListCollectiveOffersResponseModel, api=blueprint.api)
+@adage_jwt_required
+def get_educational_offers_by_uai_from_algolia(
+    authenticated_information: AuthenticatedInformation,
+) -> serializers.ListCollectiveOffersResponseModel:
+    if authenticated_information.uai is None:
+        raise ApiErrors({"institutionId": "institutionId is required"}, status_code=400)
+
+    list_offers_by_uai = educational_repository.get_all_collective_offers_by_institutionUAI(
+        uai=authenticated_information.uai
+    )
+
+    for offer in list_offers_by_uai:
+        serialized_favorite_offers = []
```

Comment:
Il faudrait sortir cette variable de la boucle, non ? Parce que là... on recréé une nouvelle liste vide à chaque itération.